LightGBM/data.py

```diff
@@ -107,7 +150,9 @@ def features(df):
     windows = [7, 28]
     for window in windows :
         for lag,lag_col in zip(lags, lag_cols):
-            df[f"rmean_{lag}_{window}"] = df[["id", lag_col]].groupby("id")[lag_col].transform(lambda x: x.rolling(window).mean())
+            df[f"rmean_{lag}_{window}"] = df[["id", lag_col]] \
```

Comment:
I also struggled with this. The solution for me was to use one line for each operation. See e.g. `_time_series_weights()` in **nn.py**. You get rid of the ugly backslashes and you can give a new name after each operation, which makes it a lot clearer.